x-pack/plugins/observability/e2e/test_reporter.ts

```diff
@@ -0,0 +1,222 @@
+    }
+    const message = `${symbols[status]} Took  (${renderDuration(end - start)} seconds)`;
+    this.write(message);
+
+    await fs.promises.mkdir('.journeys/failed_steps', { recursive: true });
+
+    await gatherScreenshots(join(CACHE_PATH, 'screenshots'), async (screenshot) => {
+      const { data, step } = screenshot;
+
+      if (status === 'failed') {
+        await (async () => {
+          await fs.promises.writeFile(join('.journeys/failed_steps/', `${step.name}.jpg`), data, {
+            encoding: 'base64',
+          });
+        })();
+      }
+    });
+  }
+
+  onEnd() {
+    const failedJourneys = Array.from(this.journeys.entries()).filter(([, steps]) =>
+      steps.some((step) => step.status === 'failed')
+    );
+
+    if (failedJourneys.length > 0) {
+      failedJourneys.forEach(([journeyName, steps]) => {
+        if (process.env.CI) {
+          const name = red(`Journey: ${journeyName} 🥵`);
+          this.write(`\n+++ ${name}`);
+          steps.forEach((stepResult) => {
+            const { status, end, start, error, name: stepName } = stepResult;
+            const message = `${symbols[status]}  Step: '${stepName}' ${status} (${renderDuration(
+              (end - start) * 1000
+            )} ms)`;
+            this.write(indent(message));
+            if (error) {
+              this.write(renderError(error));
+            }
+          });
+        }
+      });
+    }
+
+    const successfulJourneys = Array.from(this.journeys.entries()).filter(([, steps]) =>
+      steps.every((step) => step.status === 'succeeded')
+    );
+
+    successfulJourneys.forEach(([journeyName, steps]) => {
+      // fs.unlinkSync('.journeys/videos/' + journeyName + '.webm');
+    });
+
+    const { failed, succeeded, skipped } = this.metrics;
+    const total = failed + succeeded + skipped;
+
+    let message = '\n';
+    if (total === 0) {
+      message = 'No tests found!';
+      message += ` (${renderDuration(now())} ms) \n`;
+      this.write(message);
+      return;
+    }
+
+    message += succeeded > 0 ? green(` ${succeeded} passed`) : '';
+    message += failed > 0 ? red(` ${failed} failed`) : '';
+    message += skipped > 0 ? cyan(` ${skipped} skipped`) : '';
+    message += ` (${renderDuration(now() / 1000)} seconds) \n`;
+    this.write(message);
+  }
+
+  write(message: any) {
+    if (typeof message === 'object') {
+      message = JSON.stringify(message);
+    }
+    log(message + '\n');
+  }
+}
+
+const SEPARATOR = '\n';
+
+function indent(lines: string, tab = '   ') {
+  return lines.replace(/^/gm, tab);
+}
+
+const NO_UTF8_SUPPORT = process.platform === 'win32';
+const symbols = {
+  warning: yellow(NO_UTF8_SUPPORT ? '!' : '⚠'),
+  skipped: cyan('-'),
+  progress: cyan('>'),
+  succeeded: green(NO_UTF8_SUPPORT ? 'ok' : '✓'),
+  failed: red(NO_UTF8_SUPPORT ? 'x' : '✖'),
+};
+
+function now() {
+  return performance.now();
+}
+
+function findPWLogsIndexes(msgOrStack: string): [number, number] {
```

Comment:
It would be better to use it from Synthetics directly than copy pasting it. 

PW might change format.